### PR TITLE
add docs saying mixed-cluster ILM is not supported

### LINKED ISSUE #37085: Document and/or gracefully fail creating invalid ILM policies in mixed-cluster
**Problem Context**

While operating a mixed-version cluster where the master node version contains new lifecycle actions 
that previously didn't exist, there can be non-friendly failures.

Creating policies with newly introduced actions will result in success on the master, but then 
can either fail when hitting ILM apis from older coordinating nodes, or running unsupported actions 
on other nodes.

Two scenarios:

1. Requesting policy from older node:

```
{"error":{"root_cause":[{"type":"transport_serialization_exception","reason":"Failed to deserialize response from handler [org.elasticsearch.transport.TransportService$Con
textRestoreResponseHandler]"}],"type":"transport_serialization_exception","reason":"Failed to deserialize response from handler [org.elasticsearch.transport.TransportServi
ce$ContextRestoreResponseHandler]","caused_by":{"type":"illegal_argument_exception","reason":"Unknown NamedWriteable [org.elasticsearch.xpack.core.indexlifecycle.Lifecycle
Action][freeze]"}},"status":500}
```

2. Executing new lifecycle action that hits APIs that do not exist in older nodes

**Solution**

There needs to be more guidance on how to best manage new policies while operating mixed clusters. It should be recommended to disable ILM while a rolling-upgrade is in progress. This would avoid these unintuitive errors.

In addition to better guidance in documentation, it may make sense to disallow creation of these invalid policies while in mixed-clusters and explain the situation to the user.

### PR DESCRIPTION
**Pull request by talevy**
Closes #37085.

### CODE REVIEW

File: docs/reference/ilm/index.asciidoc [hunk 1]
```diff
@@ -46,6 +46,16 @@ to a single shard.
 . After 7 days, move the index into the cold stage and move it to less expensive
 hardware.
 . Delete the index once the required 30 day retention period is reached.
+
+[IMPORTANT]
+===========================
+{ilm} does not support mixed-cluster usage. Usage of new policies using
+new actions that do not exist in the oldest version of a cluster with
+mixed-versioned nodes. Although it may be possible to create such new
```
Comment (gwbrown) [inline docs/reference/ilm/index.asciidoc]: This sentence doesn't have a verb - this should say something more like (new text in brackets): 
```
Usage of new policies using
new actions that do not exist in the oldest version of a cluster with
mixed-versioned nodes [will cause errors if those actions are run on nodes which do not support those new actions].
```

Comment (gwbrown) [reply docs/reference/ilm/index.asciidoc]: To be clear, that may not be what we want to say, but as written the sentence is confusing.

Comment (talevy) [reply docs/reference/ilm/index.asciidoc]: haha woops. edited super last minute and didn't fully read it

**Review by AthenaEryma — APPROVED** [review]
LGTM

**Pull request merged**